packages/sdk-js/src/GrowthBook.ts

```diff
@@ -488,12 +498,48 @@ export class GrowthBook<
 
     // Apply new changes
     if (result.inExperiment) {
-      const undo = this._applyDOMChanges(result.value);
-      if (undo) {
-        this._activeAutoExperiments.set(experiment, {
-          undo,
-          valueHash,
-        });
+      if (result.value.urlRedirect && experiment.urlPatterns) {
+        const url = experiment.persistQueryString
+          ? mergeQueryStrings(this._getContextUrl(), result.value.urlRedirect)
+          : result.value.urlRedirect;
+
+        if (isURLTargeted(url, experiment.urlPatterns)) {
+          this.log(
+            "Skipping redirect because original URL matches redirect URL",
+            {
+              id: experiment.key,
+            }
+          );
+          return result;
+        }
+        this._redirectedUrl = url;
+        const navigate = this._getNavigateFunction();
+        if (navigate) {
+          if (isBrowser) {
+            this._setAntiFlicker();
+            window.setTimeout(() => {
+              try {
+                navigate(url);
+              } catch (e) {
+                console.error(e);
```

Comment:
Could consider having an unset function for antiFlicker and not having a timeout for situations with a slow network. i.e. navigate takes a long time and antiflicker is removed because the 3.5 second default timeout passes